## 2.3 Protein Structure

###### 18.05.2017 \| [Slides](https://www.rostlab.org/sites/default/files/fileadmin/teaching/SoSe17/PP1CS/20170518_PP1_poteins.pdf) \| [Wiki](https://i12r-studfilesrv.informatik.tu-muenchen.de/sose17/pp4cs1/index.php/Protein_structures)

---

#### 1. Keywords

##### Amino acids, side chains, residues

* 20 different amino acids
* **Backbone:** all of them have the same backbone \(see following picture\)
* **Side Chain:** the differ in their side-chains, which also give them unique features. 
  * electrically charged side chains \(positive or negative\)
  * polar uncharged side chains
  * hydrophobic side chains
  * special cases
* Amino Acids are the building blocks of proteins. Once it is part of a protein, they amino acids is referred to as a protein

![](/assets/Peptidformationball.svg.png)

##### Protein sequence

* linear sequence of amino acids \(residues\), connected by their backbone
* formed by a condensation reaction \(formation of a peptide bond, under creation of a water molecule\)
* oriented from **N-Terminus** to **C-Terminus**
* typically starts with **Methionine \(AUG codon\)**

##### Secondary, tertiary, quaternary protein structure

* **Secondary Structure**
  * local structure elements: **⍺-helix, β-sheet, loops**
  * building blocks for higher order structures
  * stabilized by **hydrogen bonds**
  * amino acids have 'preferences' for certain secondary structure elements

| Coding / Representation | Protein Aspect |
| :--- | :--- |
| **1D Information**: sequence of amino acids as a string | **Primary Structure**: amino acid sequence |
| **2D Information:** 2D-Array, contact map | **Secondary Structure:** helices, sheets, ... |
| **3D Information:** coordinates or atom couplings | **Tertiary Structure:** Spatial arrangement of secondary structure elemets |

##### 

Hydrogen bonds

Alpha helix, beta sheet, loop, random coil, disordered region

Protein features: hydrophobicity, solvent accessibility, active site, binding site

Ramachandran plot

Protein domain, PFAM \([http://pfam.xfam.org\](http://pfam.xfam.org%29\)

Protein Data Bank \(PDB\)

Root mean square deviation \(RMSD\)

Protein function

#### 2. Exercises

##### 2.1 Questions

**Question:** What are the building blocks of proteins?

> ?

**Question:** Define protein backbone and amino acid side chain in 1 or 2 sentences for each term.

> ?

**Question:** How many amino acids appear in proteins? How can they be classified?

> ?

**Question:** Name atom types involved in a hydrogen bond. Do S-H groups form hydrogen bonds? Why \(not\)?

> ?

**Question:** How are alpha helices held together?

> ?

**Question: **What is similar and what is different in the hydrogen bonding of the alpha helix and the beta sheet?

> ?

**Question: **Why do we find "forbidden" areas in a Ramachandran plot?

> ?

**Question:** What is a protein domain?

> ?

**Question:** How many amino acids are typically found a in a domain? Why is there a minimum/maximum size?

> ?

##### 2.2 PDB

**Question:** How many structures are stored in the PDB? How many of those are protein structures?

> ?

**Question:** Which experimental methods are \(mainly\) used to determine the structures? How long does it on average take to find the 3D structure for one protein for each method?

> ?

**Question:** How many human \("Homo sapiens"\) protein structures are in the PDB?

> ?

**Question:** Why does the number of protein structures already decreases when reducing at 100% sequence identity? Why does it decrease when reducing at even lower sequence identity further?

> ?

##### 2.3 Molecular Visualization

```markdown
# Note: Neither mentioned in exercises nor lectures. Let's hope there is no question regarding this.
```



