## 3.1 Lecture Questions

> This section contains possible exam questions asked Professor Rost in the lectures he dedicated to answering student questions. They are **highly relevant, **because he will sample exam questions from this pool.

### Questions

**Question: **With a matching \_profile-profile** **\_comparison what can you say about the two families?

> The assumption is that matching profiles share a similar / same structure and function.

**Question: **When I build a profile of a famliy: Do they share the same structure? Should I verify that they do? How do i do that?

> The very assumption is that the proteins of one family share the same structure and function. When iteratively refining the profile with proteins retrieved by _profile-sequence_ of \_profile-profile \_comparison \(from the twilight- / midnight-zone\), it can make sense to double check the new proteins with secondary structure prediction to avoid adding false-positives.

**Question: **Cross-Validation: What is it? How does it work? Why do we need it?

**Question: **What is the difference between a BLOSUM matrix and a PSSM \(Position Specific Substitution Matrix\)?

**Question: **What is the most successful method to predict 3D structure?

**Question: **What is homology modeling \(= comparative modeling\) and how does it work? What are the limitations of it?

**Question: **How can you predict structure in the \[a\] daylight- \[b\] twilight- \[c\] midnight-zone?**Question: **What is the assumption behind all alignment methods that is incorrect and nevertheless seems to work? Give a method that aligns 2 proteins without that assumption.

> The assumption is that the alignment of the residue at position **i **is independent of the **i+1**. \(Short alignment i and i+1 are independent\).  
> ❓ Method w/o that?

**Question: **Why do we have so few experimentally confirmed structures in the PDB?

> ca. 85 million proteins are known, but only about 120 000 are in the PDB.
>
> Sequencing technology has improved and become a lot cheaper and hence many more proteins were discovered. While there were improvements in 3D structure determination, it still costs at least 100 000 euros to experimentally determine the 3D structure of a proteins



